src/Microsoft.Net.Http.Headers/MediaTypeHeaderValue.cs

```diff
@@ -53,13 +53,13 @@ public string Charset
                 HeaderUtilities.ThrowIfReadOnly(IsReadOnly);
                 // We don't prevent a user from setting whitespace-only charsets. Like we can't prevent a user from
                 // setting a non-existing charset.
-                var charsetParameter = NameValueHeaderValue.Find(_parameters, CharsetString);
+                var charsetParameter = NameValueHeaderValue.Find(ParametersInternal, CharsetString);
```

Comment:
`Find()` returns `null` if the first parameter is `null`. So this could now unnecessarily allocate an `ObjectCollection` instance.